### DISCUSSION TITLE
Include images in collection

### DISCUSSION THREAD

**Discussion by hatamiarash7** [jekyll/jekyll#9153] [Q&A] [answered]
I have custom collections for my Jekyll project:

```YAML
collections_dir: events
collections:
  event1:
    output: true
    permalink: /:collection/:title
    sort_by: date
  event2:
    output: true
    permalink: /:collection/:title
    sort_by: date
```

And it's my directory structure:

```text
project
├── _config.yml
└── events
    ├── _event1
    │   ├── images
    │   │   ├── title1.jpg
    │   │   └── title2.jpg
    │   ├── title1.md
    │   └── title2.md
    └── _event2
        ├── images
        │   ├── title3.jpg
        │   └── title4.jpg
        ├── title3.md
        └── title4.md
```

- I have two events collection
- Every collection has some docs
- Every doc needs an image

### Problem

Currently, I should manually put all images in the `assets` folder so I can use them like this `/assets/img/event1/title1.jpg` :

```
project
├── _config.yml
├── events
└── assets
    └── img
        └── event1
            ├── title1.jpg
            ├── title2.jpg
            ├── ...
            └── ...
```

It's very hard to manage image assets for every collection because I have many of them. Is there any solution that every collection has its own `asset` folder ( Like the first directory tree ) and Jekyll's build process copies them in the final `_site` directory?

Unfortunately, this doesn't work. Jekyll will pick **one image of every extension** from every collection, rename it to the collection's name ( `event1.jpg` / `event1.png` ) and copy that to the site's root and return an error message for other ones:

```
Conflict: The following destination is shared by multiple files.
          The written file may end up with unexpected contents.
          /<project>/_site/event1.jpg
           - /<project>/events/_event1/images/title2.jpg
           - /<project>/events/_event1/images/title3.jpg
           - /<project>/events/_event1/images/title4.jpg
```

I don't understand this behavior. Do you have any idea how to do this?

**Comment by ashmaroli** [comment, marked as answer]
Hello @hatamiarash7 
Static files are automatically included for custom collections (any collection except `posts`).
The reason behind the awkward behavior you've encountered is because of your `permalink` configuration for your collections. The permalink config applies to both documents and asset files in a collection. Static files do not have `:title` placeholders defined. So, they are ignored resulting in the two image files getting assigned same permalink "value" (as proved by the conflict warning).

**Reply by hatamiarash7** [reply]
Is there any way to solve this? What can I do? Do you have any idea?

**Reply by ashmaroli** [reply]
Have you tried building without defining the `permalink` keys for the two collections in the config file?

**Reply by hatamiarash7** [reply]
I tried to build with this permalink: `/:collection/:name`. This will solve the problem.  
By using this method, the final names are not pretty, but it doesn't matter, I can solve this problem in other ways.

Thank you, your comment gave me a new idea and solved the problem 🙏
